api/v1alpha1/workspace_validation.go

```diff
@@ -89,7 +89,31 @@ func (w *Workspace) validateUpdate(old *Workspace) (errs *apis.FieldError) {
 	return errs
 }
 
-func (r *TuningSpec) validateCreate(ctx context.Context, workspaceNamespace string) (errs *apis.FieldError) {
+func (r *AdapterSpec) validateCreateorUpdateAdapter() (errs *apis.FieldError) {
+	if r.Source == nil {
+		errs = errs.Also(apis.ErrMissingField("Source"))
+	} else {
+		errs = errs.Also(r.Source.validateCreateDataSource().ViaField("Adapters"))
+
+		if r.Source.Name == "" {
```

Comment:
The name must be a valid DNS subdomain value (used as the container name). Please do corresponding check for the validity.